lib/lazy_object.rb

```diff
@@ -27,4 +27,17 @@ def __target_object__
   def method_missing(method_name, *args, &block)
     __target_object__.send(method_name, *args, &block)
   end
+
+  def present?
+    !blank?
+  end
+
+  def blank?
+    __target_object__.respond_to?(:empty?) ? !!__target_object__.empty? : !__target_object__
+  end
+
+  # Forwards method calls to the target object.
```

Comment:
I think it should be a proxy object as far as it's possible.

Instead of checking `@__target_object__.` for presence would it make to add `@__loaded__` boolean attribute and to set it to true once target object is loaded?